### DISCUSSION TITLE
Check all rows on a table ?

### DISCUSSION THREAD

**Discussion by t-prod** [livewire/flux#472] [Feature Requests]
Hi,

I try to simply check all ? in a <flux:table>
This component could be the solution : <flux:checkbox.group> but I cannot make it work with this DOM :

```
<flux:table :paginate="$this->results">
        <flux:checkbox.group>
        <flux:columns>
        </flux:columns>
        <flux:rows>
             <flux:cell><flux:checkbox /></flux:cell>
       </flux:rows>
       </flux:checkbox.group>
</flux:table>
```

How can you doi it ?

**Comment by jeffchown** [comment]
@t-prod Not sure if you have access to Caleb's Livewire screencasts, but if you do, he made 2 screencasts that may be helpful:
- https://livewire.laravel.com/screencast/data-tables/bulk-actions
- https://livewire.laravel.com/screencast/data-tables/adding-select-all

**Comment by t-prod** [comment]
You're right I have access to these screencasts I've forgot everything is explained for this feature ;)
Thanks

**Reply by jeffchown** [reply]
You're welcome :)

**Reply by richardvil** [reply]
@t-prod how you solved this issue, do you implemented some custom code from the screencasts? I've tried to use the flux.checkbox.group with flux.checkbox.all but it seems like a hack or workaround.

**Comment by alxndrmlr** [comment]
Chiming in here as  we were discussing this in the Livewire discord and @joshhanley suggested I add my +1 re: making this native in Flux.

I had also watched Caleb's screencasts, and we lifted from them and abstracted it into a reusable blade component / trait. 

It'd be great to get the 1st party touch on this :D 

FWIW for others this is what I've written and used in my projects https://gist.github.com/alxndrmlr/a64c28f8b8e492a84de7ffd9ab883fc6
https://gist.github.com/alxndrmlr/5be147af78b62a6d0327f17bd5c07c01

**Reply by JellyBool** [reply]
Hi @alxndrmlr 

Would you mind telling me where `{{$key}}` comes from:

```blade
<div x-data="selectAll('{{ $key }}')">
    <flux:checkbox x-on:change="handleCheck" x-ref="checkbox" />
</div>
```

**Reply by alxndrmlr** [reply]
```blade
<flux:table :paginate="$this->rows">
    <flux:table.columns>
        <flux:table.column><x-select-all key="userIdsOnPage"></x-select-all></flux:table.column>
        <flux:table.column sortable :sorted="$sortBy === 'name'" :direction="$sortDirection" wire:click="sort('name')">{{ __('Name') }}</flux:table.column>
        <flux:table.column sortable :sorted="$sortBy === 'title'" :direction="$sortDirection" wire:click="sort('title')">{{ __('Title') }}</flux:table.column>
        <flux:table.column sortable :sorted="$sortBy === 'email'" :direction="$sortDirection" wire:click="sort('email')">{{ __('Email') }}</flux:table.column>
        <flux:table.column sortable :sorted="$sortBy === 'last_login_at'" :direction="$sortDirection" wire:click="sort('last_login_at')">{{ __('Last Login') }}</flux:table.column>
        <flux:table.column>{{ __('Role') }}</flux:table.column>
        <flux:table.column>{{ __('Groups') }}</flux:table.column>
        <flux:table.column></flux:table.column>
    </flux:table.columns>
    <flux:table.rows>
        @foreach($this->rows() as $user)
                ...
        @endforeach
    </flux:table.rows>
</flux:table>
```

```php

<?php

// ... Truncated

class Page extends Component
{
    use AuthorizesRequests;
    use HasFluxTable;

    #[Locked]
    public Organization $organization;

    #[Locked]
    public array $userIdsOnPage = [];

    #[Url(except: null)]
    public $groups = [];

    #[Url(except: null)]
    public $roles = [];

    public function mount()
    {
        //...
    }
}
```

@JellyBool $key is a prop that is name of the of the attribute on my component that holds the selectedIds. This likely could be abstracted into the `HasFluxTable` trait as well

**Reply by JellyBool** [reply]
Got it! Thank you @alxndrmlr

**Comment by PrinceMinky** [comment]
Wrap the whole table and it should work. I got this to work:

```

<flux:checkbox.group>
<flux:table :paginate="$this->results">
        <flux:table.columns>
          <flux:table.column>
           <flux:checkbox.all />
          </flux:table.column>
        </flux:table.columns>
        <flux:table.rows>
        <flux:table.row>
          <flux:table.cell>
            <flux:checkbox />
          </flux:table.cell>
        </flux:table.rows>
        <flux:table.row>
          <flux:table.cell>
            <flux:checkbox />
          </flux:table.cell>
        </flux:table.rows>
       </flux:table.rows>
</flux:table>
</flux:checkbox.group>
```
